doc/users/next_whats_new/fig_draw_no_output.rst

```diff
@@ -0,0 +1,8 @@
+Figure now has draw_no_output method
+------------------------------------
+
+Rarely, the user will want to trigger a draw without making output to 
+either the screen or a file.  This is useful for determining the final 
+position of artists on the figure that require a draw like text.
+This could be accomplished via ``fig.canvas.draw()`` but that is 
+not user-facing, so a new method on `.Figure.draw_no_output` is provided.  
```

Comment:
I don't agree that `fig.cavas.draw()` is not user facing, we use it through out the examples and tutorials!  I think the benefit here is that `draw_on_output` does not have any side effects out side of the `Figure` (where as `draw` may update a GUI or in the case of (iirc) the SVG backend require a file to be open), is marginally faster (due to the `renderer.draw_*` methods being stubbed out to no-ops), and is "at the top level" on the `Figure` object.

Reply:
Fair enough.  All I meant by that is unless you are making a GUI or a backend you never need to know about the canvas.   

I guess to suggest a heuristic to resolve the tension above: if we want users to know about and actively use an object in the normal course of things, we should guide them to it by removing top-level methods.  In the case of "axis" it is arguably useful for the user to know about the axis objects and manipulate them directly, so it's probably worth removing some of the shadowing methods on "axes".   For canvas I don't think we need any users to call methods on the canvas, unless of course they are making a GUI, so in this case it makes sense to bubble the method up.  

I'll change the wording (definitely fewer side effects is another great reason to switch), look into moving the implementation to the public method.  

We do use fig.canvas.draw a lot, so some follow up PRs would be great.  Probably a good first issue activity.

Reply:
New text:
```
Rarely, the user will want to trigger a draw without making output to 
either the screen or a file.  This is useful for determining the final 
position of artists on the figure that require a draw like text.
This could be accomplished via ``fig.canvas.draw()`` but has side effects,
sometimes requires an open file, and is documented on an object most users 
do not need to access.  The `.Figure.draw_no_output` is provided to trigger 
a draw without pushing to the final output with fewer side effects.
```